src/Couchbase.Lite.Shared/API/Document/Blob.cs

```diff
@@ -282,6 +283,34 @@ internal Blob([NotNull]Database db, [NotNull]IDictionary<string, object> propert
 
         #endregion
 
+        #region Public Methods
+
+        /// <summary>
+        /// Return whether the given dictionary represents Blob or not
+        /// </summary>
+        /// <param name="blobDict"> JSON Dictionary represents in the <see cref="Blob"/> :
+        /// Key          | Value                  | Mandatory | Description
+        /// ---------------------------------------------------------------------------------------------------
+        /// @type        | constant string “blob” | Yes       | Indicate Blob data type.
+        /// content_type | String                 | No        | Content type ex. text/plain.
+        /// length       | Number                 | No        | Binary length of the Blob in bytes.
+        /// digest       | String                 | Yes       | The cryptographic digest of the Blob’s content.
+        /// </param>
+        /// <returns>Return true if the given dictionary represents Blob, otherwise return false</returns>
+        public static bool IsBlob(Dictionary<string, object> blobDict)
+        {
+            if (!blobDict.ContainsKey(Constants.ObjectTypeProperty) || (string) blobDict[Constants.ObjectTypeProperty] != Constants.ObjectTypeBlob
+                || (blobDict.ContainsKey(Blob.ContentTypeKey) && blobDict[Blob.ContentTypeKey].GetType() != typeof(string))
+                || (blobDict.ContainsKey(Blob.LengthKey) && blobDict[Blob.LengthKey].GetType() != typeof(int))
+                || blobDict[Blob.DigestKey].GetType() != typeof(string)) {
```

Comment:
in case of a Blob which is not saved to DB, this might fail right? since the digest is not present?

Reply:
From the public API point of view, digest is required.

Reply:
The GetBlob API (line 974) will return null if the digest info doesn't existed as expected in spec. If I add digest null checking into IsBlob, which returns bool, and GetBlob throws exception if IsBlob returns false. No digest will returns false in isBlob, we then will not be able to get null result from GetBlob when there is no digest existed.